namespaceFinder.php

```diff
@@ -11,20 +11,35 @@ class namespaceFinder {
     private $wantedNs;
     private $isSafe;
 
+    /**
+     * Resolves the namespace on construction
+     * 
+     * @param string $path the namespace link
+     */
     function __construct($path){
         $this->wantedNs = $this->computeWantedNs($path);
         $this->sanitizeNs();
     }
 
     private function computeWantedNs($path){
         global $ID;
+        global $conf;
+        
+        // Convert all other separators to colons
+        if ($conf['useslash']) $path = str_replace('/', ':', $path);
```

Comment:
I'm not sure I completely understand that line, or more precisely, whether we should test against that option. I mean: here `$path` is provided by the user, so I guess the question is whether we want to allow a syntax like `<nspages my/ns>`. If we do, then I think there is no ambiguity when this syntax is used (because, according to https://www.dokuwiki.org/pagename ; slashes are not allowed in page name, so if a user puts a `/` it can only mean that it's intended to be a namespace separator).
To put it in a nutshell: if we want to allow such syntax, shouldn't we support it regardless of the value of that conf option?

Reply:
The short answer is that, that's just how dokuwiki implements it for page links.

For example if `useslash` *is not set* then the page link `my/ns` is resolves to `my_ns` (because it goes through cleanID). But if it *is set*, it resolves to `my:ns`.

But since nspages doesn't call cleanID and assumes the user input is valid, I suppose that it makes sense to always treat slashes as namespace separators.

Btw nspages, as it is, happens to already handles "/" in paths. That's because the slashes are never removed nor escaped before it is converted to a filesystem path.